### PR TITLE
Handle model outputs and missing values in import

### PR DESCRIPTION
**Pull request by tsalemink**
"Model Outputs" (Angles, Forces, Moments, Powers, Scalars) are now filtered out of the list of point labels when importing C3D data. C3D files that have had post-processing done will automatically have these values included in the list of point labels, but most of these should not be included in the TRC data. [Commit: [Filter out C3D model outputs from marker labels](https://github.com/hsorby/trc-data-reader/commit/32f2407f56ad9bf34d33b7cb3d8b71132844fbc5)]

The way the first data header line is constructed has been updated. A call to `strip` the line has been removed as this was removing tabs from the end of the line and these tabs are required for lining up the data-header and data-value columns. [Commit: [Include tabs at the end of marker label header](https://github.com/hsorby/trc-data-reader/commit/d377d38d6440298c76ab867c1f0781e3a14b673d)]

The `_import_from_c3d` method has been updated to handle C3D files with missing coordinates. Missing coordinates are now written as empty strings when writing TRC data. [Commit: [Write missing coordinates as empty strings](https://github.com/hsorby/trc-data-reader/commit/eb93600936e46f92f705a248e8e46981db0c9db0)]

### CODE REVIEW

File: src/trc.py [hunk 1]
```diff
@@ -195,23 +200,45 @@ def _import_from_c3d(self, filename):
...
+            point_group = reader.get('POINT')
+            model_outputs = set()
+            for param in ['ANGLES', 'FORCES', 'MOMENTS', 'POWERS', 'SCALARS']:
+                if param in point_group.param_keys():
+                    model_outputs.update(point_group.get(param).string_array)
+            point_labels = []
+            for param in ['LABELS', 'LABELS2']:
+                if param in point_group.param_keys():
+                    filtered_labels = [None if label in model_outputs else label.strip() for label in point_group.get(param).string_array]
+                    point_labels.extend(filtered_labels)
```
Comment (hsorby) [inline src/trc.py]: This should be added as a keyword argument so users can override and change this list of filtered data.  Something like
```
filter_output=None
if filter_output is None:
  filter_output = ['ANGLES', 'FORCES', 'MOMENTS', 'POWERS', 'SCALARS']
```

same for `params`.

File: src/trc.py [hunk 2]
```diff
@@ -259,7 +286,7 @@ def save(self, filename):
 
             for frame in self['Frame#']:
                 time, line_data = self[frame]
-                values = [f'{v:.5f}' for values in line_data for v in values]
+                values = ['' if math.isnan(v) else f'{v:.5f}' for line_data in line_data for v in line_data]
```
Comment (hsorby) [inline src/trc.py]: I don't think having `for line_data in line_data` is very good.

**Review by hsorby — CHANGES_REQUESTED** [review]
I couple of small changes to improve the code.

File: src/trc.py (hunk 1 above)
Comment (tsalemink) [reply src/trc.py]: Just added this.

File: src/trc.py (hunk 2 above)
Comment (tsalemink) [reply src/trc.py]: Oops! Thanks for catching that.

File: src/trc.py [hunk 3]
```diff
@@ -195,32 +200,59 @@ def _import_from_c3d(self, filename):
...
+                                break
+                        point_data.append(coordinates)
+                self[i] = time, point_data
+
+    def import_from(self, filename, *args, **kwargs):
         """
         Import data from a non-TRC file source.
         Currently, the only alternative supported format is c3d.
 
         :param filename: The source file of the data to be imported.
```
Comment (hsorby) [inline src/trc.py]: Just need to document those keyword arguments here that you can now use, fancy.

**Review by hsorby — CHANGES_REQUESTED** [review]
A little helpful documentation please.

File: src/trc.py (hunk 3 above)
Comment (tsalemink) [reply src/trc.py]: I've just added that documentation. Let me know if you want me to add/remove/change anything.

**Review by hsorby — APPROVED** [review]
Nice, I think we'll go with this.

**Pull request merged**